File: backend/app/schema/property.py
import enum

from pydantic import BaseModel, ConfigDict


class PropertyType(enum.Enum):
    HOUSE = "house"
    APARTMENT = "apartment"


class PropertyCreate(BaseModel):
    address: str
    location: str
    type: PropertyType
    characteristics: str | None = None


class PropertyResponse(BaseModel):
    id: int
    address: str
    location: str
    type: PropertyType
    characteristics: str | None = None

    model_config = ConfigDict(from_attributes=True)


class PropertyUpdate(BaseModel):
    address: str | None = None
    location: str | None = None
    type: PropertyType | None = None
    characteristics: str | None = None

File: frontend/src/types/property.ts
import type { ReviewResponse } from '@/types/review';

export type PropertyType = "house" | "apartment";

export interface PropertyCreate {
  property_id: string;
  address: string;
  location: string;
  type: PropertyType;
  characteristics: string | null;
}

export interface PropertyResponse {
  id: number;
  property_id: string;
  address: string;
  location: string;
  type: PropertyType;
  characteristics: string | null;
}

export interface PropertyDetails {
  id: number;
  title: string;
  location: string;
  price: number;
  image: string;
  type: string;
  realEstateName: string;
  characteristics: string | null;
  averageRating?: number | null;
  reviews: ReviewResponse[];
}

export interface PropertyUpdate {
  address?: string;
  location?: string;
  type?: PropertyType;
  characteristics?: string | null;
}